src/components/absorber/module.cairo

```diff
@@ -0,0 +1,402 @@
+        }
+        fn get_final_time(self: @ContractState, slot: u256) -> u64 {
+            let times = self._times.read(slot);
+            if times.len() == 0 {
+                return 0;
+            }
+            times[times.len() - 1]
+        }
+        fn get_times(self: @ContractState, slot: u256) -> Span<u64> {
+            self._times.read(slot).array().span()
+        }
+        fn get_absorptions(self: @ContractState, slot: u256) -> Span<u64> {
+            self._absorptions.read(slot).array().span()
+        }
+        fn get_absorption(self: @ContractState, slot: u256, time: u64) -> u64 {
+            let times = self._times.read(slot).array();
+            if times.len() == 0 {
+                return 0;
+            }
+
+            let absorptions = self._absorptions.read(slot).array();
+            if absorptions.len() == 0 {
+                return 0;
+            }
+
+            interpolate(
+                time,
+                times.span(),
+                absorptions.span(),
+                Interpolation::Linear(()),
+                Extrapolation::Constant(())
+            )
+        }
+        fn get_current_absorption(self: @ContractState, slot: u256) -> u64 {
+            self.get_absorption(slot, get_block_timestamp())
+        }
+        fn get_final_absorption(self: @ContractState, slot: u256) -> u64 {
+            let absorptions = self._absorptions.read(slot);
+            if absorptions.len() == 0 {
+                return 0;
+            }
+            absorptions[absorptions.len() - 1]
+        }
+        fn get_project_value(self: @ContractState, slot: u256) -> u256 {
+            self._project_value.read(slot)
+        }
+        fn get_ton_equivalent(self: @ContractState, slot: u256) -> u64 {
+            self._ton_equivalent.read(slot)
+        }
+        fn is_setup(self: @ContractState, slot: u256) -> bool {
+            self._project_value.read(slot)
+                * self._times.read(slot).len().into()
+                * self._absorptions.read(slot).len().into()
+                * self._ton_equivalent.read(slot).into() != 0
+        }
+        fn set_absorptions(
+            ref self: ContractState,
+            slot: u256,
+            times: Span<u64>,
+            absorptions: Span<u64>,
+            ton_equivalent: u64
+        ) {
+            // [Check] Times and prices are defined
+            assert(times.len() == absorptions.len(), 'Times and absorptions mismatch');
+            assert(times.len() > 0, 'Inputs cannot be empty');
+            assert(ton_equivalent > 0, 'Ton equivalent must be positive');
+
+            // [Effect] Clean times and absorptions
+            let mut stored_times = self._times.read(slot);
+            let mut stored_absorptions = self._absorptions.read(slot);
+            let mut index = stored_times.len();
+            loop {
+                if index == 0 {
+                    break;
+                }
+                stored_times.pop_front();
```

Comment:
is it necessary to empty the List first instead of creating a new one?

Reply:
hum, good point it could works

Reply:
I cleaned list using `len = 0`